src/Raven.Client/Documents/Linq/LinqPathProvider.cs

```diff
@@ -636,5 +642,18 @@ public static bool IsIncludeCall(MethodCallExpression mce)
         {
             return mce.Method.DeclaringType == typeof(RavenQuery) && mce.Method.Name == nameof(RavenQuery.Include);
         }
+
+        private static bool TryUnwrapImplicitOperatorIfNeeded(Expression expression, out Expression newExpression)
+        {
+            newExpression = expression;
+
+            if (expression is MethodCallExpression { Method.Name: "op_Implicit" } callExpression)
```

Comment:
Can you please elaborate more on this implicit operator handling? Why do we have to start dealing with that?

Reply:
This is due to how this is getting resolved in the expression tree. When MemoryExtensions are used instead of EnumerableExtensions, then first parameter gets wrapped in `op_Implicit` method call, I'm just unwrapping it, since I do not think we care about that operation to understand the expression. 
So for `string[] MyArray` we are getting: `op_Implicit(string[] MyArray)`